### PR TITLE
BUG: lib: Handle axes with length 0 in np.unique.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
This PR is an alternative to gh-15565.  It avoids the big block of special-case code in that PR.

### CODE REVIEW

File: numpy/lib/arraysetops.py [hunk 1]
```diff
@@ -270,20 +270,24 @@ def unique(ar, return_index=False, return_inverse=False,
...
     ar = np.ascontiguousarray(ar)
     dtype = [('f{i}'.format(i=i), ar.dtype) for i in range(ar.shape[1])]
 
     try:
-        consolidated = ar.view(dtype)
+        if ar.shape[1] > 0:
+            consolidated = ar.view(dtype)
+        else:
+            consolidated = np.empty(len(ar), dtype=dtype)
     except TypeError:
```
Comment (eric-wieser) [inline numpy/lib/arraysetops.py]: This `if` could probably do with a brief comment explaining why the special case is needed

**Review by eric-wieser — APPROVED** [review]
This looks great, thanks for looking for a tidier solution

**Comment by tylerjereddy** [comment]
@seberg @WarrenWeckesser There's the reason--these comment triggers are pretty handy.

**Comment by tylerjereddy** [comment]
@chrispat Any ideas here? We haven't changed our YML file recently.

**Comment by chrispat** [comment]
@vtbassmatt have there been any updates in path filtering?

**Comment by WarrenWeckesser** [comment]
@tylerjereddy, @chrispat: Looks like we're not the only ones: https://stackoverflow.com/questions/60677704/why-did-azure-pipelines-suddenly-stopped-appearing-in-pr-on-github-open-source

**Comment by tylerjereddy** [comment]
Indeed, cross-linked SciPy PR above seems to fail at the moment too.

**Comment by vtbassmatt** [comment]
We're having an incident at the moment. Engineering is engaged.

File: numpy/lib/arraysetops.py (hunk 1 above)
Comment (WarrenWeckesser) [reply numpy/lib/arraysetops.py]: I pushed an update that added comments about this `if` statement.  Copy-editing suggestions are welcome!

**Comment by mattip** [comment]
@vtbassmatt is there a place we can monitor progress on the incident?

**Comment by vtbassmatt** [comment]
https://status.dev.azure.com/

**Comment by mattip** [comment]
Thanks @WarrenWeckesser

**Pull request merged**